### PR TITLE
Rating widget accessibility

### LINKED ISSUE #280: dojox.form.Rating needs accessible interaction and semantics
The current rating widget uses a list with no accessible text or keyboard interaction.

Both the W3C [example](https://www.w3.org/WAI/tutorials/forms/custom-controls/#a-star-rating) and many other accessible rating widgets use radio buttons as a base. The existing hidden input could be swapped out for a set of radio buttons corresponding to the number of stars.

### PR DESCRIPTION
**Pull request by smhigley**
Updates the rating widget to use radio inputs for each star to add proper semantics and keyboard accessibility.

Resolves #280

### CODE REVIEW

File: form/Rating.js [hunk 1]
```diff
@@ -68,16 +78,31 @@ define([
...
-			var hoverValue = +domAttr.get(evt.target, "value");
+			var hoverValue = +domAttr.get(evt.target.querySelector('input'), "value");
 			this._renderStars(hoverValue, true);
 			this.onMouseOver(evt, hoverValue);
 		},
 
+		_onClick: function(evt) {
+			if (evt.target.tagName === 'LABEL') {
+				var clickedValue = +domAttr.get(evt.target.querySelector('input'), "value");
+				// for backwards compatibility
```
Comment (dylans) [inline form/Rating.js]: backwards compatibility with previous versions of Dojo or browsers or ?

Comment (smhigley) [reply form/Rating.js]: For previous versions of Dojo, because there used to be a `value` attribute directly on the `<li>` tag

Comment (dylans) [reply form/Rating.js]: That's fine, the comment should say that so we remember what type of compatibility we were trying to preserve. :)

**Comment by dylans** [comment]
Thanks @smhigley!

Closed via 8b7a975. Backported as b454122 (1.13), ee5ee6c (1.12), bdca832 (1.11), 592f137 (1.10), 6967e62 (1.9), and a6a4666 (1.8).

**Pull request closed without merging**